### ISSUE TITLE
Prerequisites fail "guess_host_provider KeyError: 'ansible_product_name'"

### ISSUE THREAD

**Issue by yyefet-chtr** [openshift/openshift-ansible#11497]
#### Description

Provide a brief description of your issue here. For example:

    "module_stderr": "Shared connection to [server] closed.\r\n", 
    "module_stdout": "AttributeError(\"'NoneType' object has no attribute 'startswith'\",)\r\nKeyError('ansible_os_family',)\r\nTraceback (most recent call last):\r\n  File \"/root/.ansible/tmp/ansible-tmp-1555192403.95-217076418340476/AnsiballZ_openshift_facts.py\", line 113, in <module>\r\n    _ansiballz_main()\r\n  File \"/root/.ansible/tmp/ansible-tmp-1555192403.95-217076418340476/AnsiballZ_openshift_facts.py\", line 105, in _ansiballz_main\r\n    invoke_module(zipped_mod, temp_path, ANSIBALLZ_PARAMS)\r\n  File \"/root/.ansible/tmp/ansible-tmp-1555192403.95-217076418340476/AnsiballZ_openshift_facts.py\", line 48, in invoke_module\r\n    imp.load_module('__main__', mod, module, MOD_DESC)\r\n  File \"/tmp/ansible_openshift_facts_payload_6ZuGLr/__main__.py\", line 1305, in <module>\r\n  File \"/tmp/ansible_openshift_facts_payload_6ZuGLr/__main__.py\", line 1292, in main\r\n  File \"/tmp/ansible_openshift_facts_payload_6ZuGLr/__main__.py\", line 1045, in __init__\r\n  File \"/tmp/ansible_openshift_facts_payload_6ZuGLr/__main__.py\", line 1067, in generate_facts\r\n  File \"/tmp/ansible_openshift_facts_payload_6ZuGLr/__main__.py\", line 1197, in init_provider_facts\r\n  File \"/tmp/ansible_openshift_facts_payload_6ZuGLr/__main__.py\", line 1136, in guess_host_provider\r\nKeyError: 'ansible_product_name'\r\n", 
    "msg": "MODULE FAILURE\nSee stdout/stderr for the exact error", 
##### Version

Please put the following version information in the code block
indicated below.

* Your ansible version per `ansible --version`
ansible --version
ansible 2.7.10
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/root/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python2.7/site-packages/ansible
  executable location = /usr/bin/ansible
  python version = 2.7.5 (default, Apr 11 2018, 07:36:10) [GCC 4.8.5 20150623 (Red Hat 4.8.5-28)]

If you're operating from a **git clone**:
openshift-ansible-3.11.105-1-6-g9d65c4c

* The output of `git describe`

If you're running from playbooks installed via RPM

* The output of `rpm -q openshift-ansible`
openshift-ansible-3.11.37-1.git.0.3b8b341.el7.noarch

Place the output between the code block below:

```
CentOS Linux release 7.5.1804 (Core)  
```

##### Steps To Reproduce
clone 3.11
run prerequisites.yml

##### Expected Results
Describe what you expected to happen.

```
Example command and output or error messages
```

##### Observed Results
Describe what is actually happening.

```
Example command and output or error messages
```

For long output or logs, consider using a [gist](https://gist.github.com/)

##### Additional Information

Provide any additional information which may help us diagnose the
issue.

* Your operating system and version, ie: RHEL 7.2, Fedora 23 (`$ cat /etc/redhat-release`)
* Your inventory file (especially any non-standard configuration parameters)
* Sample code, etc

```
EXTRA INFORMATION GOES HERE
```

**Comment by yyefet-chtr** [comment]
Resolved by downgrading Ansible to 2.6.0 and prereqs passed.

**Comment by denis-pingin** [comment]
Same issue here, resolved by downgrading Ansible to version 2.7.9. It seems that "ansible_product_name" fact is not collected anymore.

**Comment by mtnbikenc** [comment]
This is fixed in https://github.com/openshift/openshift-ansible/pull/11516.

There was a change in Ansible 2.8 devel which was backported to stable-2.7 and landed in 2.7.10.

See https://github.com/ansible/ansible/issues/55400.
